### DISCUSSION TITLE
[FEATURE REQUEST] Generate and rotate SSH keys for a specific user across the minion hosts

### DISCUSSION THREAD

**Discussion by ajs95** [saltstack/salt#67256] [Feature Requests]
**Describe the solution you'd like**
Require a module to configure passwordless ssh for specific user. 
To generate and rotate SSH keys for a specific user across the minion hosts. 

**Describe alternatives you've considered**
Alternative solutions was to run Shell scripts with "cmd.run" module

**Please Note**
If this feature request would be considered a substantial change or addition, this should go through a SEP process here https://github.com/saltstack/salt-enhancement-proposals, instead of a feature request.

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html)
  - [Join our Community Slack](https://via.vmw.com/salt-slack)
  - [IRC on LiberaChat](https://web.libera.chat/#salt)
  - [Salt Project YouTube channel](https://www.youtube.com/channel/UCpveTIucFx9ljGelW63-BWg)
  - [Salt Project Twitch channel](https://www.twitch.tv/saltprojectoss)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at saltproject@vmware.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!

**Comment by dwoz** [comment]
> **Please Note** If this feature request would be considered a substantial change or addition, this should go through a SEP process here https://github.com/saltstack/salt-enhancement-proposals, instead of a feature request.

We no longer maintain the SEP process.

**Comment by nf-brentsaner** [comment]
@ajs95 Unless I'm missing something, this can already be done easily using a cmd.run for ssh-keygen and a [scheduler](https://docs.saltproject.io/salt/user-guide/en/latest/topics/scheduler.html). Or by just adding and enabling a [systemd timer](https://www.freedesktop.org/software/systemd/man/latest/systemd.timer.html).

What does this FR add that either of those doesn't accomplish?

**Comment by lkubb** [comment]
I'm unsure whether this is a duplicate of https://github.com/saltstack/salt/issues/65197 or which functionality is requested in addition to that.

> this can already be done easily using a cmd.run for ssh-keygen and a scheduler

While `cmd.run` can serve as a workaround for missing SSH key management ability, it's often brittle. Many included modules are essentially a more portable/reliable version of their CLI counterpart...

> adding and enabling a systemd timer

... and serve to integrate the functionality into Salt more deeply.

The PR fixing the above-mentioned FR (https://github.com/saltstack/salt/pull/64708) should help with the implementation of a robust and configurable scheduled key rotation SLS. It would also allow to go one step further and have a basic Salt-managed SSH CA\*, which avoids having to rely on the scheduler/mine/reactors.

\* like it is already possible with X509 certs
